src/main/webapp/app/exercises/text/participate/text-editor.component.ts

```
@@ -255,16 +336,25 @@
-        setLatestSubmissionResult(this.submission, getLatestSubmissionResult(this.submission));
+        const submissionToCreateOrUpdate = this.submission;
+        // id undefined creates a new submission and setting results to undefined prevents foreign key constraints when deleting results from submission
+        if (this.hasAthenaResultForLatestSubmission) {
+            submissionToCreateOrUpdate.id = undefined;
+            submissionToCreateOrUpdate.results = undefined;
+        } else {
+            setLatestSubmissionResult(submissionToCreateOrUpdate, getLatestSubmissionResult(this.submission));
+        }
 
-        this.textSubmissionService.update(this.submission, this.textExercise.id!).subscribe({
+        this.textSubmissionService.update(submissionToCreateOrUpdate, this.textExercise.id!).subscribe({
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)